src/klein/test/typing_app.py

```diff
@@ -71,3 +80,36 @@ def returnsDeferredTag(self, request: IRequest) -> KleinRenderable:
     @router.route("/deferred-none")
     def returnsDeferredNone(self, request: IRequest) -> KleinRenderable:
         return succeed(None)
+
+    # Ensure that same return object types for a route are valid in async
+    # methods.
+
+    @router.route("/async-object")
+    async def asyncReturnsObject(self, request: IRequest) -> KleinRenderable:
+        return succeed(object())  # type: ignore[arg-type]
```

Comment:
This is wrong because coroutines should never return `Deferred` objects.  The correct thing would be `return await succeed(...)` but … there's not really any point to that :)

Reply:
oh right yes sure